### ISSUE TITLE
Vendor complete set of migrate drivers

### ISSUE THREAD

**Issue by nyarly** [goharbor/harbor#16420]
I'm trying to address #8649 and finding that, while I can reproduce the `pg_advisory_lock` issue in tests, when I try to switch the migrator to use `cockroachdb` I run into trouble. It seem that the `database` directory in `golang-migrate/migrate` has been stripped back to just `postgresql`.

Would you please restore the `database` directory of that package? Or at least `database/cockroachdb`?

Thanks very much.

**Comment by nyarly** [comment]
@wy65701436 @stonezdj I'm not sure I understand "help wanted" on this issue. Per `CONTRIBUTING.md`, core maintainers are responsible for maintaining the vendor directory, and contributors shouldn't change module dependencies.

Even so, I was experimenting with changing the module vendoring just to see if I could close the issue, and found that:
1. I couldn't figure out how modules were being vendored in the first place. There doesn't seem to be a script or other tool for managing it,
2. The existing `go.mod` doesn't seem to be legitimate; removing the vendor directory prevents the project from building
3. Manually adding `cockroach` from a fresh download of the `golang-migrate/migrate` module isn't sufficient; there's other dependencies that brings in.

In short, what help can the community provide on this issue?

**Comment by wwentland** [comment]
Is there a chance this could be re-opened and addressed? It would be great if:

- Harbor could include the complete API shipped with https://github.com/golang-migrate/migrate
- Adopt a standard way of managing dependencies that allows for community contributions.

While the second point might be less straightforward to implement, the first one should be trivial.  

What would be the best way to move this forward, @wy65701436 @stonezdj  ?
